pkg/crc/preflight/preflight_checks_network_linux.go

```diff
@@ -72,9 +72,29 @@ server=/crc.testing/192.168.130.11
 dns=dnsmasq
 `
 
-	crcNetworkManagerOldDispatcherPath = filepath.Join(crcNetworkManagerRootPath, "dispatcher.d", "pre-up.d", "99-crc.sh")
-	crcNetworkManagerDispatcherPath    = filepath.Join(crcNetworkManagerRootPath, "dispatcher.d", "99-crc.sh")
-	crcNetworkManagerDispatcherConfig  = `#!/bin/sh
+	crcNetworkManagerOldDispatcherPath      = filepath.Join(crcNetworkManagerRootPath, "dispatcher.d", "pre-up.d", "99-crc.sh")
+	crcNetworkManagerDispatcherPath         = filepath.Join(crcNetworkManagerRootPath, "dispatcher.d", "99-crc.sh")
+	crcNetworkManagerDispatcherLegacyConfig = `#!/bin/sh
+# This is a NetworkManager dispatcher script to configure split DNS for
+# the 'crc' libvirt network.
+#
+# The corresponding crc bridge is not created through NetworkManager, so
+# it cannot be configured permanently through NetworkManager. We make the
+# change directly using systemd-resolve instead.
+#
+# NetworkManager will overwrite this systemd-resolve configuration every time a
+# network connection goes up/down, so we run this script on each of these events
+# to restore our settings. This is a NetworkManager bug which is fixed in
+# version 1.26.6 by this commit:
+# https://cgit.freedesktop.org/NetworkManager/NetworkManager/commit/?id=ee4e679bc7479de42780ebd8e3a4d74afa2b2ebe
+
+export LC_ALL=C
+
+systemd-resolve --interface crc --set-dns 192.168.130.11 --set-domain ~testing
```

Comment:
Ideally we'd be able to use this command both on older and newer systems? Any idea if this would work? The introduction of `crcNetworkManagerDispatcherLegacyConfig` deserves to be split in its own commit though.

Reply:
`resolvectl` supports being called as `systemd-resolve` (which on RHEL 8 is a symlink to `resolvectl`, and same with `resolvconf`) and will convert its arguments to the necessary compatibility points. The native syntax type for `resolvectl` is the new verb based CLI.

https://github.com/systemd/systemd/blob/9813d8354cc9af33b739e8d478368adfb548edc1/src/resolve/resolvectl.c#L3398-L3403

However, that only applies to 239+ where the new binary change was made. In distributions using 238- there is no translation to the older syntax, so you have to use the original CLI. The options I've used here are my best guess to what the original syntax would have been.

So, hypothetically we could use the old `systemd-resolve` as the official call, which would work (assuming my command is correct) on older distributions like Ubuntu 18.04, and on newer systems using 239+ as it'll get translated properly under the hood by `resolvectl`.

Reply:
Sorry for the multi-comments, these things come to me as I look at the code. The reason I went with the multi-config approach is because I didn't find a comparable argument set for the `resolvectl default-route crc false`.

Reply:
I'm not sure `default-route` is important, it's one of these "should not hurt to have this" addition, but I'd say having it or not should not make a difference.

Reply:
Ubuntu 20.04 has `resolvctl` and `set-default`. The 'resolvconf' executable from the `resolvconf` package is a shell script, not sure it's related to systemd at all.

Reply:
Looking at it's sources, it doesn't appear to be. `resolvectl` has the ability to act as a passthrough for `resolvconf`, which it is in the Red Hat family. Though it looks like Canonical isn't following suite on that just yet.

I'll get some eyes on my `systemd-resolve` command and check to make sure it's an accurate representation of what's the `resolvectl` runs are doing. The other option is to just not support systemd-resolved if `resolvectl` isn't available (something I was playing with originally).

Reply:
I did a quick test with a Fedora 33 Server install running the aforementioned `systemd-resolve` command (swapped `--interface` with a different device) and it seemed to produce the output matching the `crc` interface managed by `resolvectl`.